src/Common/Survey/SurveyHelper.cs

```diff
@@ -0,0 +1,335 @@
+﻿using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.WindowsAzure.Commands.Common.Survey
+{
+    public class SurveyHelper
+    {
+        private const int _countExpiredDays = 30;
+        private const int _lockExpiredDays = 30;
+        private const int _surveyTriggerCount = 3;
+        private const int _flushFrequecy = 5;
+        private const int _delayForSecondPrompt = 2;
+        private const int _delayForThirdPrompt = 5;
+
+        private SurveyHelper()
+        {
+            CurrentDate = DateTime.UtcNow.ToString("yyyy'-'MM'-'dd");
+            IgnoreSchedule = "Disabled".Equals(Environment.GetEnvironmentVariable(AzurePowerShell.AzurePSInterceptSurvey));
```

Comment:
We might want to be more tolerant to casing?
```suggestion
            IgnoreSchedule = "Disabled".Equals(Environment.GetEnvironmentVariable(AzurePowerShell.AzurePSInterceptSurvey, StringComparsion.xxxIgnoreCase));
```